moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js

```diff
@@ -1,69 +1,62 @@
 import React from "react";
-import { useState } from "react";
-import { CircularProgress } from "@material-ui/core";
+import { CircularProgress, TextField } from "@material-ui/core";
 import { Autocomplete, Alert } from "@material-ui/lab";
 import { useSocrataGeojson } from "src/utils/socrataHelpers";
 import {
-  useSignalChangeEffect,
   getSignalOptionLabel,
   getSignalOptionSelected,
-  useInitialSignalComponentValue,
-  renderSignalInput,
   SOCRATA_ENDPOINT,
 } from "src/utils/signalComponentHelpers";
 import { filterOptions } from "src/utils/autocompleteHelpers";
 
 /**
  * Material Autocomplete wrapper that enables selecting a traffic/phb signal record from a
  * Socrata dataset and setting it as a project component.
- * * @param {Object} setSelectedComponentSubtype - sets the selectedComponentSubtype from parent state
- * * @param {Object} setEditFeatureCollection - sets the editFeatureCollection from parent state
- * * @param {Boolean} editFeatureCollection - the geoJSON generated for all the the features in this component
- * * @param {Object} classes - MaterialUI style object
- *  @return {JSX.Element}
+ * @param {Object} classes - MaterialUI style object
+ * @param {Function} onChange - callback function to run when the signal is changed for React Hook Form
+ * @param {Object} value - the signal feature to set as the value of the autocomplete from React Hook Form
+ * @return {JSX.Element}
  */
-const SignalComponentAutocomplete = ({
-  classes,
-  setSelectedComponentSubtype,
-  setEditFeatureCollection,
-  editFeatureCollection,
-}) => {
-  const [signal, setSignal] = useState(null);
-  const { features, loading, error } = useSocrataGeojson(SOCRATA_ENDPOINT);
+const SignalComponentAutocomplete = React.forwardRef(
+  ({ classes, autocompleteProps, onChange, value }, ref) => {
+    const { features, loading, error } = useSocrataGeojson(SOCRATA_ENDPOINT);
```

Comment:
Because we've done away with the "Subtype" field that used to exist on this form, it's actually possible to select the "Signal - PHB" component but choose a traffic signal (and vice versa). That's an oversight on my part and not a big deal—but i will make a backlog issue to filter the autocomplete options by signal type.

<img width="606" alt="Screen Shot 2023-01-04 at 2 13 03 PM" src="https://user-images.githubusercontent.com/14793120/210631553-e0244f5e-bdf0-4233-86ae-8ebe10273a39.png">

Reply:
- https://github.com/cityofaustin/atd-data-tech/issues/11115

Reply:
Awesome - I initially filtered this way but then was confused about the lack of school zone beacon and rrfd types. Forgot to follow up on that when we synced on what should/should not use the signal dropdown. Long way to say that it should be a small effort to put that back in. 👍